magefile.go

```diff
@@ -0,0 +1,256 @@
+//go:build mage
+// +build mage
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+var (
+	tools = []string{
+		"github.com/bufbuild/buf/cmd/buf",
+		"github.com/bufbuild/buf/cmd/protoc-gen-buf-breaking",
+		"github.com/bufbuild/buf/cmd/protoc-gen-buf-lint",
+		"github.com/golangci/golangci-lint/cmd/golangci-lint",
+		"github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway",
+		"github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2",
+		"golang.org/x/tools/cmd/cover",
+		"golang.org/x/tools/cmd/goimports",
+		"google.golang.org/grpc/cmd/protoc-gen-go-grpc",
+		"google.golang.org/protobuf/cmd/protoc-gen-go",
+	}
+
+	Default = Build
+)
+
+// Bootstrap installs tools required for development
+func Bootstrap() error {
+	fmt.Println("Bootstrapping tools...")
+	if err := os.Chdir("_tools"); err != nil {
+		return fmt.Errorf("failed to change dir: %w", err)
+	}
```

Comment:
One option, where you have these `Chdir` calls, is to use `exec.Cmd` directly.
You can set the `.Dir` field accordingly on the `Cmd` structure. To save changing the processes context back and forth.

Take it or leave it.

Reply:
always forget that `exec.Cmd` can do this